crates/uv-pep440/src/version.rs

```diff
@@ -1416,6 +1418,39 @@ impl FromStr for VersionPattern {
     }
 }
 
+/// Lifetime and indexing workaround to allow accessing the release as `&[u64]` even though the
+/// digits may be stored in a compressed representation.
+pub struct Release<'a> {
+    inner: ReleaseInner<'a>,
+}
+
+enum ReleaseInner<'a> {
+    // The small versions unpacked into larger u64 values.
+    // We're storing at most 4 u64 plus determinant for the duration of the release call on the
+    // stack, without heap allocation.
+    Small0([u64; 0]),
+    Small1([u64; 1]),
+    Small2([u64; 2]),
+    Small3([u64; 3]),
+    Small4([u64; 4]),
+    Full(&'a [u64]),
+}
```

Comment:
I was wondering if there was a more compact representation than this, so I tried:

```rust
enum ReleaseInner<'a> {
    Small { numbers: [u64; 4], len: u8 },
    Full(&'a [u64]),
}
```

But indeed, it's the same size (which makes sense).